src/onnx/parse_multi_head_attention.cpp

```diff
@@ -0,0 +1,295 @@
+                          const std::vector<instruction_ref>& args) const
+    {
+        if(not contains(info.attributes, "num_heads"))
+            MIGRAPHX_THROW("MultiHeadAttention: num_heads attribute is required");
+
+        int64_t num_heads = parser.parse_value(info.attributes.at("num_heads")).at<int>();
+
+        multi_head_attention_parameters params;
+        check_inputs(args, num_heads, params);
+
+        auto query = args[0];
+        instruction_ref key;
+        instruction_ref value;
+
+        if(params.qkv_fomat == qkv_fomat_t::qkv_packed)
+        {
+            // Packed QKV: (batch_size, q_sequence_length, num_heads, 3, head_size)
+            unpack_qkv(info, query, key, value);
+        }
+        else
+        {
+            // Query: (batch_size, q_sequence_length, hidden_size)
+            std::vector<int64_t> q_dims{
+                params.batch_size, params.q_sequence_length, num_heads, params.head_size};
+            query = info.add_instruction(make_op("reshape", {{"dims", q_dims}}), query);
+
+            key = args[1];
+
+            if(params.qkv_fomat == qkv_fomat_t::kv_packed)
+            {
+                // Packed KV: (batch_size, kv_sequence_length, num_heads, 2, head_size)
+                unpack_kv(info, key, value);
+            }
+            else
+            {
+                value = args[2];
+                if(params.qkv_fomat == qkv_fomat_t::q_k_v)
+                {
+                    // Key: (batch_size, kv_sequence_length, hidden_size)
+                    // Value: (batch_size, kv_sequence_length, hidden_size_v)
+                    std::vector<int64_t> k_dims{
+                        params.batch_size, params.kv_sequence_length, num_heads, params.head_size};
+                    std::vector<int64_t> v_dims{params.batch_size,
+                                                params.kv_sequence_length,
+                                                num_heads,
+                                                params.head_size_v};
+                    key   = info.add_instruction(make_op("reshape", {{"dims", k_dims}}), key);
+                    value = info.add_instruction(make_op("reshape", {{"dims", v_dims}}), value);
+                }
+            }
+        }
+
+        // Target shape: (batch_size, num_heads, sequence_length, head_size)
+        std::vector<int64_t> perm{0, 2, 1, 3};
+        query = info.add_instruction(make_op("transpose", {{"permutation", perm}}), query);
+        if(params.qkv_fomat != qkv_fomat_t::q_k_v_cross)
+        {
+            key   = info.add_instruction(make_op("transpose", {{"permutation", perm}}), key);
+            value = info.add_instruction(make_op("transpose", {{"permutation", perm}}), value);
+        }
+
+        float scale = 1 / std::sqrt(params.head_size);
+        if(contains(info.attributes, "scale"))
+            scale = parser.parse_value(info.attributes.at("scale")).at<float>();
```

Comment:
Add a case where you have scale not default. I didn't see that in your test cases